### ISSUE TITLE
Can we suggest an import for a new name, in a Quick Fix?

### ISSUE THREAD

**Issue by sethladd** [dart-atom/dartlang#730]
I tried Control-1 on a name that wasn't imported, but is available from a library provided by a package.

I saw these options:

![screen shot 2016-01-26 at 4 27 38 pm](https://cloud.githubusercontent.com/assets/5479/12599823/dac791ea-c449-11e5-9ea5-c198e6f75b7a.png)

Could we add an option, like "Import package:flutter/widget.dart" ?

Atom 1.4.1, dartlang 0.5.3, SDK 1.14.0-dev.7.2 running on macos.

**Comment by devoncarew** [comment]
@scheglov, is something like this already handled by the analysis server - and something I need to wire up - or would this be a feature request?

**Comment by scheglov** [comment]
If the library `package:flutter/widget.dart` is know to the analysis context, i.e. it is analyzer because any other library (implicit or explicit) imports it, then the Analysis Server will produce the import quick fix. If it not used, and exists only as an imported package in `pubspec.yaml` - we know nothing about it, and don't produce the quick fix.

**Comment by sethladd** [comment]
Hi @scheglov it comes from a package in my pubspec.yaml. The analyzer doesn't scan all dependencies to make them potentially available?

**Comment by scheglov** [comment]
No, analyzer does not look at code which is not in the project and is not referenced from a project library.

**Comment by lukechurch** [comment]
@sethladd I had a prototype a while back that would use a service to index the places you might have got this from. So you could write a type and it would offer the various packages that export it. It this is important enough, and there is agreement to from the other folks, I can fairly easily bring it back to life and productionise...

**Comment by sethladd** [comment]
The baby step here might be "look in all packages referenced by .packages/pubspec.lock". Don't wait for the library to be manually imported (even transitively). I've already done the work to run `pub get`... the library is sitting right there to be imported :)

cc @kevmoo FYI
